play-json/jvm/src/test/scala/play/api/libs/json/JsonSpec.scala

```diff
@@ -60,66 +83,369 @@ class JsonSpec extends org.specs2.mutable.Specification {
+          Json.parse(
+            s"""
+               |{
+               |  "float": 123.123,
+               |  "double": 1
+               |}""".stripMargin)
+            .as[FloatNumbers].float mustEqual 123.123f
+        }
+
+        "success for valid negative number" in {
+          Json.parse(
+            s"""
+               |{
+               |  "float": -123.123,
+               |  "double": 1
+               |}""".stripMargin)
+            .as[FloatNumbers].float mustEqual -123.123f
+        }
+
+        "success for max value" in {
+          val outOfLimits = BigDecimal(Float.MaxValue.toString)
+          Json.parse(
+            s"""
+               |{
+               |  "float": $outOfLimits,
+               |  "double": 1
+               |}""".stripMargin)
+            .as[FloatNumbers].float mustEqual Float.MaxValue
+        }
+
+        "success for min value" in {
+          val outOfLimits = BigDecimal(Float.MinValue.toString)
+          Json.parse(
+            s"""
+               |{
+               |  "float": $outOfLimits,
+               |  "double": 1
+               |}""".stripMargin)
+            .as[FloatNumbers].float mustEqual Float.MinValue
+        }
+      }
+
+      "for Double" should {
+
+        "success for valid positive number" in {
+          Json.parse(
+            s"""
+               |{
+               |  "float": 1,
+               |  "double": 123.123
+               |}""".stripMargin)
+            .as[FloatNumbers].double mustEqual 123.123d
+        }
+
+        "success for valid negative number" in {
+          Json.parse(
+            s"""
+               |{
+               |  "float": 1,
+               |  "double": -123.123
+               |}""".stripMargin)
+            .as[FloatNumbers].double mustEqual -123.123d
+        }
+
+        "success when parsing max value" in {
+          val outOfLimits = BigDecimal(Double.MaxValue)
+          Json.parse(
+            s"""
+               |{
+               |  "float": 1,
+               |  "double": $outOfLimits
+               |}""".stripMargin)
+            .as[FloatNumbers].double mustEqual Double.MaxValue
+        }
+
+        "success when parsing min value" in {
+          val outOfLimits = BigDecimal(Double.MinValue)
+          Json.parse(
+            s"""
+               |{
+               |  "float": 1,
+               |  "double": $outOfLimits
+               |}""".stripMargin)
+            .as[FloatNumbers].double mustEqual Double.MinValue
+        }
+      }
+
+      "for BigDecimals" should {
+        "maintain precision when parsing BigDecimals within precision limit" in {
+          val n = BigDecimal("12345678901234567890.123456789")
+          val json = toJson(n)
+          parse(stringify(json)) mustEqual json
+        }
+
+        "truncate when exceeding the precision limit" in {
+          // last two six are exceeding 34 precision limit
+          val n = BigDecimal("10.12345678912345678912345678912345666")
```

Comment:
Use `0` on the `123456789` so it's easier to count.

Reply:
Done.